compiler/qsc_passes/src/spec_gen.rs

```diff
@@ -35,6 +35,30 @@ pub enum Error {
     #[error(transparent)]
     AdjGen(adj_gen::Error),
 
+    #[error("invalid specialization generator")]
+    #[diagnostic(code("Qsc.SpecGen.InvalidAdjGen"))]
+    #[diagnostic(help(
+        "valid specialization generators for adjoint are `auto`, `invert`, and `self`"
+    ))]
+    InvalidAdjGen(#[label] Span),
+
+    #[error("invalid specialization generator")]
+    #[diagnostic(code("Qsc.SpecGen.InvalidBodyGen"))]
+    #[diagnostic(help("body specialization only supports `intrinsic`"))]
```

Comment:
Is `intrinsic` only valid for `body` or can other specializations be `intrinsic`? And do we need a check to make sure that if `body` is `intrinsic` the other specializations are either not explicitly listed or are also `intrinsic`?

Reply:
Intrinsic is only valid for body, and only if there are no other specializations. There are existing test cases that cover this, which is why they aren't in the diff.